openmdao/core/implicitcomponent.py

```diff
@@ -26,6 +26,15 @@ def __init__(self, **kwargs):
         """
         super(ImplicitComponent, self).__init__(**kwargs)
 
+        if overrides_method('guess_nonlinear', self, ImplicitComponent):
```

Comment:
just curious why the overrides_method check for guess_nonlinear is in __init__ but the one for apply_linear is in configure...